polygeist/tools/cgeist/driver.cc

```diff
@@ -983,7 +983,7 @@ int main(int argc, char **argv) {
     eraseHostCode(*module);
     module.get()->setAttr(mlir::gpu::GPUDialect::getContainerModuleAttrName(),
                           Builder.getUnitAttr());
-  } else
+  } else if (!deviceModule.body().empty())
```

Comment:
I'd remove this ([see](https://github.com/intel/llvm/pull/6916#discussion_r984671680)). Also, we might introduce some declarations by default that we don't want to leak in case no SYCL code is found.

Reply:
ok this can be handled in a separate change